proposed/cli-syntax-guideline.md

```diff
@@ -0,0 +1,294 @@
+* Comfort: users should relax with little fear of making a mistake 
+
+These goals are supported by:
+
+* Explicitness in context, actions and arguments
+* Generally consistency
+* Command parsers libraries
+* Minimal traps
+
+Good CLIs use these techniques in support of the goals, but there can be grey areas and a bit of messiness. Tricky and difficult to name situations sometimes means the problem isn't yet clear or too much is being done. In truly challenging naming scenarios, strict adherence to rules is less important than the goals. 
+
+## Form
+
+The general form of a command is:
+
+```terminal
+$ dotnet [context] [action] [options] [context specification] [action type] [arguments]
+```
+
+Examples from the .NET Core CLI:
+
+```terminal
+$ dotnet [project] add [options] [<PROJECT_NAME>] package <PACKAGE_ID>
+$ dotnet tool install [options] <PACKAGE_ID>
+```
+
+In these examples, `project` denotes the actual word "project" and is optional because it is the core, or default context.
+
+| Item                  | Example 1      | Example 2    | Comments                                       |
+|-----------------------|----------------|--------------|------------------------------------------------|
+| context               | project        | tool         | `project` generally omitted since core context |
+| action                | add            | install      |                                                |
+| context specification | <PROJECT_NAME> |              |                                                |
+| action type           | package        |              |                                                |
+| argument              | <PACKAGE_ID>   | <PACKAGE_ID> |                                                |
+
+
+### Context and Core Context
+
+Each command works with a core object or concept. This concept may be implicit or explicit. For example, the Docker command `build` builds an image. The context is `image`. For Docker, the core context is `image` and by default commands work on images. Docker allows the context to optionally be included so these two commands are equivalent:
+
+``` terminal
+$ docker build ...
+$ docker image build ...
+```
+
+The core context for the .NET Core CLI is the project. Like many other CLIs, this concept is a little fluffy and you will probably comfortably think about it as _the main thing I am working on right now_. This will generally be a project, but it might in some cases be a solution or a directory. We think of a solution as a super-project and in some cases a project is represented by its directory. The .NET Core CLI will adopt the Docker style of allowing the core context to be explicit or implicit because it makes help much easier. `dotnet --help` may have 30 entries while `dotnet project --help` may have 15. For example, the following commands continue to work as before:
+
+``` terminal
+$ build
+$ new
+$ publish
+$ restore
+$ run
```

Comment:
This should be `dotnet run` as opposed to `run`, right?

Reply:
Yes, thank you for catching it!!!